### PR TITLE
feat: add cli to ama-sdk core

### LINKED ISSUE #902: [Feature]: Remove `scripts` folder from SDK shell generation
### Context

Currently the SDK Shell generator create a folder `scripts` in the new SDK.
The purpose of this folder is to provide scripts required to the SDK build.
There are several issue with this way:
- The scripts generated are with `.js` extension which can be an issue in a `esm` context
- The scripts may need to be on Otter side instead of delegated to the each SDKs
- These scripts are fixing issues that may not be needed anymore thank to the build process evolutions

### Proposal

- [x] Check if the scripts are still needed
- [x] Move the scripts (that can be migrated) to `@ama-sdk/core`
- [x] Change the extension of remaining scripts to `.cjs`
- [x] Provide `ng update` to `@ama-sdk/schematics`

### PR DESCRIPTION
**Pull request by kpanot**
## Related issues

- :rocket: Feature close #902

### CODE REVIEW

File: packages/@ama-sdk/schematics/schematics/ng-update/typescript/v10.0/script-removal.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+};
+
+/**
+ * Add the CPY dependencies
+ */
+export const addCpyDependencies = (): Rule => async (tree, context) => {
+  const amaSdkSchematicsPackageJson = await readPackageJson();
+  context.addTask(new DevInstall({ packageName: `cpy@${amaSdkSchematicsPackageJson.devDependencies!.cpy as string}`}));
+  context.addTask(new DevInstall({ packageName: `cpy-cli@${amaSdkSchematicsPackageJson.devDependencies!['cpy-cli'] as string}` }));
+  return () => tree;
```
Comment (fpaul-1A) [inline packages/@ama-sdk/schematics/schematics/ng-update/typescript/v10.0/script-removal.ts:55]:
```suggestion
  return noop;
```

File: packages/@ama-sdk/core/cli/clear-index.cts [hunk 2]
```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env node
+
+/*
+ * Remove deleted models' exports
+ */
+
+import { readdir, statSync, unlink } from 'node:fs';
```
Comment (fpaul-1A) [inline packages/@ama-sdk/core/cli/clear-index.cts:7]: couldn't you use `node:fs/promises` instead ?

File: packages/@ama-sdk/core/cli/files-pack.cts [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+  const copyToDist = (file: string) => {
+    const distFile = path.resolve(baseDir, distFolder, path.relative(baseDir, file));
+    // eslint-disable-next-line no-console
+    console.log(`${file} copied to ${distFile}`);
+    return copyFileSync(file, distFile);
+  };
+
+  // Move files into the dist folder
+  const copies = files.reduce((acc, glob) => {
```
Comment (fpaul-1A) [inline packages/@ama-sdk/core/cli/files-pack.cts:68]: could be a `.map` I believe

**Review by fpaul-1A — DISMISSED** [review]

File: packages/@ama-sdk/core/cli/files-pack.cts [hunk 4]
```diff
@@ -1,16 +1,22 @@
...
-const { sync } = require('glob');
-const fs = require('node:fs').promises;
-const promisify = require('util').promisify;
+#!/usr/bin/env node
+
+/*
+ * Update the Typescript SDK Package to expose the sub modules
+ */
+
+// import * as cpy from 'cpy';
```
Comment (matthieu-crouzet) [inline packages/@ama-sdk/core/cli/files-pack.cts:7]: comment could be removed

**Review by matthieu-crouzet — APPROVED** [review]

**Review by fpaul-1A — APPROVED** [review]

**Review by fpaul-1A — APPROVED** [review]

**Review by matthieu-crouzet — APPROVED** [review]

**Pull request merged**